desci-labs/nodes
# Make build system

This PR does a couple of things:
- Further decouples the two repos
- Wraps as much of the "scripty" build steps as possible in Make which gives us nice control over the execution DAG
- Adds `clean` and `sterile` build steps to reset back to known clean state when things go haywire
- Cleans up quite a bit of bash stuff to make failures obvious and aid debugging
- Adds some healthchecks and stricter deps in the compose file to make `dockerDev.sh` more stable

# Testing procedure
1. Checkout this branch
2. Run `make clean` (you can see what it does in the root `Makefile` first)
3. Run `./dockerDev.sh`  (this automatically runs `make` and sets up all projects in the repo) 
4. Run `make sterile` (note this will nuke `local-data` too, and will `make clean` as well)
5. Run `./dockerDev.sh` again

# Feedback
Hit me with all types of feedback. Do you vibe with this? Anything that is non-obvious and need more docs? Note that I haven't yet gotten to the `README.md`.
